### DISCUSSION TITLE
How to automatically rotate DB password for Vercel

### DISCUSSION THREAD

**Discussion by clam61** [vercel/vercel#8521] [Help]
https://vercel.com/guides/how-to-allowlist-deployment-ip-address

> To properly secure your database, we recommend using a randomly generated password, stored as an [environment variable](https://vercel.com/docs/concepts/projects/environment-variables), at least 32 characters in length, and to rotate this password on a regular basis.

Could someone point me in the direction of how to set up a system where the environment variable for the DB password changes on Vercel automatically, as well as the settings for my database?  

Any recommendations for rotation frequency?  Also, do we just accept the query failures that will occur during the asynchronous DB password updates?

Assume my database is some service running on AWS

**Comment by amyegan** [comment]
You can always [update them manually](https://vercel.com/docs/concepts/projects/environment-variables#searching-and-filtering), of course, but there are also some options to automate the process. Your method of generating the new password for your DB will depend on what service you are using and your desired workflow, but you have a couple of options for updating your environment variable once you have the new password. 

You can [edit an environment variable via Vercel API](https://vercel.com/docs/rest-api#endpoints/projects/edit-an-environment-variable) or [replace and environment variable via Vercel CLI](https://vercel.com/docs/cli#commands/env). The CLI is popular for use with [GitHub Actions](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel) which would allow you to [trigger the workflow](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) based on an event or at a scheduled time.
